controllers/storagecluster/storageconsumer.go

```diff
@@ -260,14 +260,32 @@ func getLocalStorageClassNames(ctx context.Context, kubeClient client.Client, st
 	for scName := range maps.Keys(storageClassNames) {
 		// TODO: store the spec as the value which allows each value to be customizable corresponding to the name,
 		// not doing now as we only have name in the storageclassspec
-		scSpec = append(scSpec, ocsv1a1.StorageClassSpec{Name: scName})
+		scSpec = append(
+			scSpec,
+			ocsv1a1.StorageClassSpec{
+				CommonClassSpec: ocsv1a1.CommonClassSpec{
+					Name: scName,
+				},
+			},
+		)
 	}
 	return scSpec, nil
 }
 
 func getLocalVolumeSnapshotClassNames(ctx context.Context, kubeClient client.Client, storageCluster *ocsv1.StorageCluster) (
 	[]ocsv1a1.VolumeSnapshotClassSpec, error) {
 
+	internalConsumer := &ocsv1a1.StorageConsumer{}
+	internalConsumer.Name = defaults.LocalStorageConsumerName
+	internalConsumer.Namespace = storageCluster.Namespace
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(internalConsumer), internalConsumer); client.IgnoreNotFound(err) != nil {
+		return nil, err
+	}
+	if internalConsumer.UID != "" {
+		// already set in upgrade_controller
```

Comment:
Day 2 and nfs snapshot class won't be set in the upgrade controller as it runs only once per consumer. 
Additionally, day 2 vsclasses won't be added to the internal consumer after the first reconcile of the storage cluster controller, as the internal consumer will be created at this point.